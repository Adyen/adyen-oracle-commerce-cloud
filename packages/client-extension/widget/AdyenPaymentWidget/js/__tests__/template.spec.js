import Widget from '../../../../__mocks__/widget'
import generateTemplate, { mockTemplate } from '../utils/tests/koTemplate'

describe('Template', () => {
    let widget
    let tmplWidget
    beforeEach(() => {
        tmplWidget = mockTemplate('Tmpl_Widget')
        widget = new Widget()
    })

    afterEach(() => {
        tmplWidget.remove()
    })

    it('should render', function() {
        const template = generateTemplate(widget)
        expect(template).toMatchSnapshot()
    })
})
