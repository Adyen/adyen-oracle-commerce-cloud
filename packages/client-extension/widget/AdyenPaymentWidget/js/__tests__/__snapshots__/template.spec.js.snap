// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template should render 1`] = `
Object {
  Symbol(dom): <body>
    <div
      class="loader-wrapper hide"
    >
      <div
        class="lds-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        style="font-weight: bold; font-size: 16px;"
      >
        Processing...
      </div>
    </div>
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-2"
      data-oc-id="panel-2"
    />
    <br />
    <legend />
    <div
      class="adyen-collapse-heading oc-panel panel-1"
      data-oc-id="panel-1"
    />
  </body>,
}
`;
